### ISSUE TITLE
Cifar10 Training Diverging

### ISSUE THREAD

**Issue by marijnfs** [pytorch/glow#2111]
When I run the Cifar10 training example, it initially seems to be learning, getting more of the training examples right, but after a few epochs it diverges. It ends up guessing 'airplane' for all examples. I have tried reducing learning speed, turning off momentum and removing the weight decay, but it doesn't seem to help. I wonder if there is something deeper that is wrong, like gradients being accumulated that should be zeroed or something like that.
Anyone have the same result with this example?

**Comment by MatzeB** [comment]
Hi, I am trying to reproduce this (on 018a598ff0a233f6e541918a5ce987325a516921). How long did you run the example for? (Even with a release glow build and the `-cpu` commandline it looks like the example will be running for days until it reaches the 100000 iterations.
Simply running it for 1-2 hours I cannot see it getting stuck at all "airplaine"...

**Comment by nadavrot** [comment]
The problem with this test is that the neural network architecture is too small. If you increase the size of the network (by adding more conv nodes with more channels) then the network will converge (but will take longer to run). Try to replace the current network with something that looks like VGG and you will see that the problem goes away.

**Comment by MatzeB** [comment]
Apart from this test having a hard time converging in general, I think there is also a problem with `-opencl`:
I tried `-opencl` now (I was using `-cpu` before because that was the fastest backend for me) and the training seems to converge a lot slower or not at all. In fact if I print the maximum value picked out at the last layer it becomes NaN after a couple iteration so I suspect there is a problem in the opencl backend.

**Comment by marijnfs** [comment]
Thanks I'll try it out!
